nengo/connection.py

```diff
@@ -18,6 +18,15 @@
 logger = logging.getLogger(__name__)
 
 
+class PrePostParam(NengoObjectParam):
+    def validate(self, conn, nengo_obj):
+        super(PrePostParam, self).validate(conn, nengo_obj)
+        if isinstance(nengo_obj, Connection):
+            raise ValueError(
+                "Connections cannot connect to or from connections. "
```

Comment:
I'd maybe drop "connections" at the start here... seem redundant; there's no other way to connect, so probably just `"Cannot connect to or from connections."` would be sufficient.